### PR TITLE
Improve server coding guidelines

### PR DESCRIPTION
**Pull request by FuchsDominik**
### Checklist
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).

### Motivation and Context

A while ago we had a discussion in the weekly meeting about when to use `var` or the actual type in Java. I was asked to state this somewhere in the guidelines

### Description

Added an additional bullet points which points what what we discussed in the meeting: If the type is clear (by having the type somehow defined in the attribute name) and writing the type would decrease readability, `var` can be used. Otherwise use the actual type so people have the information which is needed

### CODE REVIEW

File: docs/dev/guidelines/server.rst [hunk 1]
```diff
@@ -54,6 +54,7 @@ Avoid code duplication. If we cannot reuse a method elsewhere, then the method i
 * Classes, methods or functions should be open for extension and closed for modification (open closed design principle).
 * Program for the interface and not for implementation, you should use interface type on variables, return types of a method or argument type of methods. Just like using SuperClass type to store object rather using SubClass.
 * The use of interface is to facilitate polymorphism, a client should not implement an interface method if its not needed.
+* Type inference: Use ``var`` when the type of the variable is clear and explicitly stating the type would decrease readability
```
Comment (FrankeLukas) [inline docs/dev/guidelines/server.rst:57]:
```suggestion
* Type inference: Use ``var`` when the type of the variable is clear and explicitly stating the type would decrease readability.
```

**Review by FrankeLukas — APPROVED** [review]
Lgtm otherwise.

File: docs/dev/guidelines/server.rst [hunk 2]
```diff
@@ -54,6 +54,10 @@ Avoid code duplication. If we cannot reuse a method elsewhere, then the method i
 * Classes, methods or functions should be open for extension and closed for modification (open closed design principle).
 * Program for the interface and not for implementation, you should use interface type on variables, return types of a method or argument type of methods. Just like using SuperClass type to store object rather using SubClass.
 * The use of interface is to facilitate polymorphism, a client should not implement an interface method if its not needed.
+* Type inference of variables: var vs. actual type:
+    * Variables with primitive types like int, long or also String should be defined with the actual type by default.
```
Comment (FlorianGlombik) [inline docs/dev/guidelines/server.rst:58]:
```suggestion
    * Variables with primitive types like int, long, or also String should be defined with the actual type by default.
```

Comment (FuchsDominik) [reply docs/dev/guidelines/server.rst]: 👍

File: docs/dev/guidelines/server.rst (hunk 1 above)
Comment (FuchsDominik) [reply docs/dev/guidelines/server.rst]: 👍

**Review by FlorianGlombik — APPROVED** [review]
Discussed the guidelines with @FuchsDominik and I think the current solution is not too restrictive but does still give a good idea when to use var and when not to do so.

**Review by FrankeLukas — APPROVED** [review]
Personally I prefer using `var` everywhere alongside variable names which make clear what we are dealing with, but this is also ok I guess.

**Review by krusche — APPROVED** [review]

**Pull request merged**